addons/mail/static/src/model/model_manager.js

```diff
@@ -206,33 +206,16 @@ class ModelManager {
         this._updateCycle(() => {
             const Model = record.constructor;
             if (!record.exists()) {
-                // Record has already been deleted.
-                // (e.g. unlinking one of its reverse relation was causal)
-                return;
+                throw Error(`Attempt to delete already deleted record ${record.localId}.`);
```

Comment:
Error message should stay exactly what is wrong, not what it was doing before error is raised.
~Attempt to~ delete => **Cannot** delete

_This comment applies to all changed related to errors in this file_